### ISSUE TITLE
Previous/Next Links are missing context

### ISSUE THREAD

**Issue by mgifford** [readthedocs/sphinx_rtd_theme#993]
### Problem

Another accessibility issue.
Link fails to be independently useful - WCAG 2.4.4 Link Purpose (In Context) (Level A)

### Reproducible Project

https://docs.readthedocs.io/en/stable/guides/platform.html
`<a href="autobuild-docs-for-pull-requests.html" class="btn btn-neutral float-right" title="Autobuild Documentation for Pull Requests" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>`

#### Expected Results

I'm not sure if you have a CSS class to hide content like many tools do, but essentially you want:
`<a href="autobuild-docs-for-pull-requests.html" class="btn btn-neutral float-right" title="Autobuild Documentation for Pull Requests" accesskey="n" rel="next">Next <span class="visibly-hidden">Page: Autobuild Documentation for Pull Requests</span><span class="fa fa-arrow-circle-right"></span></a>`

**Comment by Blendify** [comment]
Would adding an aria-label here be sufficient? `aria-label="Next Page"`?

**Comment by mgifford** [comment]
Well the problem is mostly that there is no context around Next. You could use [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute), but ideally it would contain the title for the string. So `aria-label="Next Page: Autobuild Documentation for Pull Requests"` or  `aria-label="Next Page: $title"` if $title were a variable name for the title of the next article.  

You're already doing essentially this for the title attribute to provide the hover effect. We just want the same affordance for assistive technology users.
